src/components/RouteSchedule/RouteSchedule.js

```diff
@@ -0,0 +1,233 @@
+
+  /**
+   * Trajectory stations informations.
+   */
+  lineInfos: PropTypes.object,
+
+  /**
+   * HTML tabIndex attribute
+   */
+  stationTabIndex: PropTypes.number,
+
+  /**
+   * Render Header of the route scheduler.
+   */
+  renderHeader: PropTypes.func,
+
+  /**
+   * Render Body of the route scheduler.
+   */
+  renderStations: PropTypes.func,
+
+  /**
+   * Function triggered on station's click event.
+   */
+  onStationClick: PropTypes.func,
+};
+
+const defaultProps = {
+  className: 'rt-route-wrapper',
+  lineInfos: null,
+  stationTabIndex: 0,
+  renderHeader: lineInfos => {
+    return (
+      <div className="rt-route-header">
+        <span
+          style={{
+            backgroundColor:
+              lineInfos.backgroundColor || bgColors[lineInfos.vehiculeType],
+            color: lineInfos.color || 'black',
+          }}
+          className="rt-route-icon"
+        >
+          {lineInfos.shortName}
+        </span>
+        <div className="rt-route-title">
+          <span className="rt-route-name">{lineInfos.destination}</span>
+          <span>{lineInfos.longName}</span>
+        </div>
+      </div>
+    );
+  },
+  renderStations: (lineInfos, stationTabIndex, onStationClick) => (
+    <div className="rt-route-body">
+      {lineInfos.stations.map((stop, idx) => (
+        <div
+          key={stop.stationId}
+          role="button"
+          className="rt-route-station"
+          onClick={e => onStationClick(stop, e)}
+          tabIndex={stationTabIndex}
+          onKeyPress={e => e.which === 13 && onStationClick(stop, e)}
+        >
+          <div className="rt-route-delay">
+            {stop.arrivalDelay ? (
+              <span
+                className={`rt-route-delay-arrival${` ${getDelayColor(
+                  stop.arrivalDelay,
+                )}`}`}
+              >
+                {`+${getDelayString(stop.arrivalDelay)}`}
+              </span>
+            ) : null}
+            {stop.departureDelay ? (
+              <span
+                className={`rt-route-delay-arrival${` ${getDelayColor(
+                  stop.departureDelay,
+                )}`}`}
+              >
+                {`+${getDelayString(stop.departureDelay)}`}
+              </span>
+            ) : null}
+          </div>
+          <div className="rt-route-times">
+            <span className="rt-route-time-arrival">
+              {getTimeString(stop.arrivalTime)}
+            </span>
+            <span className="rt-route-time-departure">
+              {getTimeString(stop.departureTime)}
+            </span>
+          </div>
+          {renderStationImg(idx, lineInfos.stations.length)}
+          <div>{stop.stationName}</div>
+        </div>
+      ))}
+    </div>
+  ),
+  onStationClick: () => {},
+};
+
+/**
+ * Displaying all stops of a line, and their informations.
+ */
+
+function RouteSchedule({
+  className,
+  lineInfos,
+  renderHeader,
+  renderStations,
+  stationTabIndex,
```

Comment:
Do we need this? I just would use `0`

Reply:
I added that to be configurable, because this library's component are aimed to be imported and used in others app. But can set 0, non configurable